File: core-network/src/main/java/com/levnovikov/core_network/api_provider/EntityConverter.java
package com.levnovikov.core_network.api_provider;

/**
 * Created by lev.novikov
 * Date: 23/2/18.
 */

public interface EntityConverter {
    <T> String convertTo(T entity);
    <T> T convertFrom(String string, Class<T> responseClass);
}

File: system-http-client/src/main/java/com/levnovikov/core_network/api_provider/EntityConverter.kt
package com.levnovikov.core_network.api_provider

import kotlin.reflect.KClass

/**
 * Created by lev.novikov
 * Date: 23/2/18.
 */

interface EntityConverter {
    fun <T : Any> convertTo(entity: T): String
    fun <T : Any> convertFrom(string: String, responseClass: KClass<T>): T
}
